### PR TITLE
Use comprehensions

### PR DESCRIPTION
**Pull request by cclauss**
### Short description

Use comprehensions

### Proposed changes

- Use list comprehensions, set comprehensions, and generators to streamline and accelerate the code.

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Review by timobrembeck — APPROVED** [review]
Cool improvements! I agree that comprehensions are better readable than the `map()` syntax.

File: tests/summ_ai_api/summ_ai_test.py [hunk 1]
```diff
@@ -231,9 +231,11 @@ async def test_translate_text_field_internal_server_error(
...
 
     errors = tuple(
-        map(
-            lambda record: record.message,
-            filter(lambda record: record.levelname == "ERROR", caplog.records),
+        (
+            record.message
+            for record in filter(
+                lambda record: record.levelname == "ERROR", caplog.records
+            )
```
Comment (david-venhoff) [inline tests/summ_ai_api/summ_ai_test.py:238]:
```suggestion
        (
            record.message
            for record in caplog.records
            if record.levelname == "ERROR"
            )
```

I think we could also get rid of the `filter` call

File: tests/summ_ai_api/summ_ai_test.py [hunk 2]
```diff
@@ -271,9 +273,11 @@ async def test_translate_text_forbidden(
...
```
Comment (david-venhoff) [inline tests/summ_ai_api/summ_ai_test.py:280]:
```suggestion
        (
            record.message
            for record in caplog.records
            if record.levelname == "ERROR"
            )
```
Same here

File: tests/summ_ai_api/summ_ai_test.py [hunk 3]
```diff
@@ -597,9 +601,11 @@ def test_check_rate_limit_exceeded(
...
```
Comment (david-venhoff) [inline tests/summ_ai_api/summ_ai_test.py:608]:
```suggestion
        (
            record.message
            for record in caplog.records
            if record.levelname == "ERROR"
            )
```
And here

**Review by david-venhoff — APPROVED** [review]
Thanks!

File: tests/summ_ai_api/summ_ai_test.py (hunk 1 above)
Comment (cclauss) [reply tests/summ_ai_api/summ_ai_test.py]: These had one closing paren too many but I fixed and squashed.

**Pull request merged**